### ISSUE TITLE
nextcloud has incorrect password requirements in inithooks

### ISSUE THREAD

**Issue by GregorKopka** [turnkeylinux/tracker#1838]
Initializing a turnkey-nextcloud:
get asked for a 8 character password with pointless requirements (see https://xkcd.com/936/),
after supplying and verifying it there comes an error message 'has to be at least 10 characters long'.

**Comment by JedMeister** [comment]
Thanks for the bug report.

As to your feedback re password requirements, whilst I'm inclined to agree in principle, prior to us adding those requirements, at least once a month (if not more often) we'd get a support request for someone who had been hacked (because the used a crappy password). Since we implemented the password requirements (many years ago now), we've only had one or 2 support requests re hacked servers - in total!

So whilst the password requirements are pointless to users who know what they are doing and don't explicitly exclude bad passwords, they are useful "guide rails" that the reduce the risk of newbs setting really bad passwords that are brute forced in seconds.

Having said that, we should probably follow Nextcloud's lead and actually compile a list of "bad passwords" to check against rather than relying on "silly" requirements.

I'm pretty sure that the primary bug is that our password function defaults to expecting 8 chars, but Nextcloud itself expects 10. The fix to that bit (i.e. ensure that it asks for 10 chars off the bat) should be pretty easy.

Unfortunately (or fortunately - depending on how you look at things) Nextcloud itself has password requirements too (beyond the length of 10 chars). So even if your password complies with our requirements (+2 char length), it still might fail.

Digging in a bit deeper, it appears that Nextcloud compiles a list of "bad passwords" from [danielmiessler/SecLists/Passords](https://github.com/danielmiessler/SecLists/tree/master/Passwords) (see [nextcloud/password_policy/lists](https://github.com/nextcloud/password_policy/tree/master/lists)). [Update: actually I found a [resource that sets out the Nextcloud password options](https://git.mdns.eu/nextcloud/passwords/-/wikis/Administrators/App-Settings#password-security-checks) and their pros and cons - whilst not directly relevant to us, it seems like a useful resource]. Nextcloud also has support for checking via the [haveibeenpwned.com API](https://haveibeenpwned.com/API/v2). That might be overkill for us, but could be a good option too?

I'm not sure when we'll look at implementing anything beyond the MVP "fix" (make it note that it requires 10 chars) but thanks again raising the issue.
